src/components/Page/components/Header/Header.tsx

```diff
@@ -135,3 +217,120 @@ export function Header({
     </div>
   );
 }
+
+function shouldShowIconOnly(
+  newDesignLanguage: boolean,
+  isMobile: boolean,
+  action: PrimaryAction,
+): PrimaryAction {
+  let {content, accessibilityLabel, icon} = action;
+  if (!newDesignLanguage || icon == null) return {...action, icon: undefined};
+
+  if (isMobile) {
+    accessibilityLabel = accessibilityLabel || content;
+    content = undefined;
+  } else {
+    icon = undefined;
+  }
+
+  return {
+    ...action,
+    content,
+    accessibilityLabel,
+    icon,
+  };
+}
+
+function notNull(value: any) {
+  return value != null;
+}
+
+function determineLayout({
```

Comment:
This and the new design language render block are the bulk of the interesting stuff in this PR. Based on a set of conditions the page component will have different layouts.